ddtrace/utils/cache.py

```diff
@@ -0,0 +1,77 @@
+from threading import RLock
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Optional
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
+
+
+miss = object()
+
+T = TypeVar("T")
+S = TypeVar("S")
+F = Callable[[T], S]
+M = Callable[[Any, T], S]
+
+
+def cached(maxsize=256):
```

Comment:
Might be worth giving a brief description of the caching strategy done here in a docstring

Reply:
Seconded 😄